### PR TITLE
SPEED-837: Remove Mobile Redirect settings override

### PR DESCRIPTION
**Pull request by aseure**
This commit removes the settings override for the [Mobile Redirect feature]() whose sunset date is set to June 15th, 2024.

After this date, it will not be able for the users of the Cloudflare dashboard, REST API or Terraform provider to change that setting, which will be automatically turned off.

Users configure [Single Redirects](https://developers.cloudflare.com/rules/url-forwarding/single-redirects/) instead of using Mobile Redirect. Refer to [Perform mobile redirects](https://developers.cloudflare.com/rules/url-forwarding/single-redirects/examples/#perform-mobile-redirects) for examples of performing mobile redirects with Single Redirects.

### CODE REVIEW

File: internal/sdkv2provider/schema_cloudflare_zone_settings_override.go [hunk 1]
```diff
@@ -202,34 +202,6 @@ var resourceCloudflareZoneSettingsSchema = map[string]*schema.Schema{
 		},
 	},
 
-	"mobile_redirect": {
```
Comment (jacobbednarz) [inline internal/sdkv2provider/schema_cloudflare_zone_settings_override.go:205]: if you're removing the schema, we'll need a schema migration here too in order to remove it from user state files. you can see some prior art in #1211

Comment (aseure) [reply internal/sdkv2provider/schema_cloudflare_zone_settings_override.go:205]: I've tried to do something like the PR you mentioned. The whole change v0 -> v1 change is just copy-pasting the value and removing the `mobile_redirect` setting from both `settings` and `initial_settings`. Let me know if there's anything else I should change.

Comment (jacobbednarz) [reply internal/sdkv2provider/schema_cloudflare_zone_settings_override.go:205]: yep! that's all that is needed here, thanks!

**Comment by jacobbednarz** [comment]
acceptance tests all passing

```
TF_ACC=1 go test ./internal/sdkv2provider -v -run "^TestAccCloudflareZoneSettingsOverride_" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareZoneSettingsOverride_Full
--- PASS: TestAccCloudflareZoneSettingsOverride_Full (34.91s)
=== RUN   TestAccCloudflareZoneSettingsOverride_RemoveAttributes
--- PASS: TestAccCloudflareZoneSettingsOverride_RemoveAttributes (46.82s)
=== RUN   TestAccCloudflareZoneSettingsOverride_NEL
--- PASS: TestAccCloudflareZoneSettingsOverride_NEL (16.17s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider	98.325s
```

File: internal/sdkv2provider/resource_cloudflare_zone_settings_override_migrate.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		return nil, fmt.Errorf("could not upgrade cloudflare_zone_settings_override from v0 to v1: state is nil")
+	}
+
+	settings, ok := rawState["settings"].(map[string]interface{})
+	if settings == nil || !ok {
+		return nil, fmt.Errorf("could not upgrade cloudflare_zone_settings_override from v0 to v1: state does not have a `settings` field")
+	}
+
+	initialSettings, ok := rawState["initial_settings"].(map[string]interface{})
+	if settings == nil || !ok {
```
Comment (jacobbednarz) [inline internal/sdkv2provider/resource_cloudflare_zone_settings_override_migrate.go:48]: the conditional here is incorrect

```suggestion
	if initialSettings == nil || !ok {
```

Comment (jacobbednarz) [reply internal/sdkv2provider/resource_cloudflare_zone_settings_override_migrate.go]: addressed in bb0417383b678572a15f272648fbe8cff4fa987f with other changes

**Pull request merged**